### PR TITLE
Add core.tanzu.vmware.com/v1alpha2 Capability resource

### LINKED ISSUE #3835: Bump core.tanzu.vmware.com Capability resource to v1alpha2
**Describe the feature request**

Since FeatureGate API changes will create new v1alpha2 API version for core.tanzu.vmware.com, we need to bump Capability resource to match that version.

The new v1alpha2 types will be identical, so we should use that as the storage version as well and not serve v1alpha1 anymore.

**Describe alternative(s) you've considered**

Don't bump the version and be out of date.

**Affected product area (please put an X in all that apply)**

- (x) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by rajathagasthya**
### What this PR does / why we need it

API group needs bumping because of the new incoming FeatureGate API changes, which means Capability resource should be at v1alpha2 as well.

This commit does the following:

* Adds new v1alpha2 Capability types which are identical to v1alpha1 types.
* Updates CRD manifest with the new version. The v1alpha1 version is no longer served and v1alpha2 is set as the storage version. Since the types are identical and we don't serve v1alpha1 anymore, we don't need a conversion webhook. The controller just reads all existing v1alpha1 resources and writes it back in the new storage version v1alpha2.
* Changes capability controller to work on v1alpha2 types.
* Updates docs.

### Which issue(s) this PR fixes

Fixes #3835 

### Describe testing done for PR

1. Install old v1alpha1 Capability CRD.
2. Run the capabilities controller and create a v1alpha1 CR:
```yaml
apiVersion: core.tanzu.vmware.com/v1alpha1
kind: Capability
metadata:
  name: tkg-capabilities
  namespace: tkg-system
spec:
  serviceAccountName: "tanzu-capabilities-manager-default-sa"
  queries:
    - name: "tanzu-cluster-with-capability"
      groupVersionResources:
        - name: "capability-resource"
          group: "core.tanzu.vmware.com"
          versions:
            - v1alpha1
          resource: "capabilities"
```
3. Update the Capability API to v1alpha2 (v1alpha1 is no longer stored) and re-run the controller.
4. Controller should re-write all existing CRs to v1alpha2.
```yaml
apiVersion: core.tanzu.vmware.com/v1alpha2
kind: Capability
metadata:
  name: tkg-capabilities
  namespace: tkg-system
spec:
  serviceAccountName: "tanzu-capabilities-manager-default-sa"
  queries:
    - name: "tanzu-cluster-with-capability"
      groupVersionResources:
        - name: "capability-resource"
          group: "core.tanzu.vmware.com"
          versions:
            - v1alpha1
          resource: "capabilities"
```
### Release note

```release-note
`capabilities.core.tanzu.vmware.com` has been updated to v1alpha2 and is identical to v1alpha1. The v1alpha1 Capability API is no longer served. Clients must immediately update to using the v1alpha2 API.
```

### CODE REVIEW

**Review by codegold79 — APPROVED** [review]
Pretty straightforward. LGTM.

File: apis/core/config/crd/bases/core.tanzu.vmware.com_capabilities.yaml [hunk 1]
```diff
@@ -326,6 +326,321 @@ spec:
             - results
             type: object
         type: object
+    served: false
+    storage: false
```
Comment (danniel1205) [inline apis/core/config/crd/bases/core.tanzu.vmware.com_capabilities.yaml:330]: question: Just wanna double check that we only want v1alpha2 to be served and stored, correct ?

Comment (rajathagasthya) [reply apis/core/config/crd/bases/core.tanzu.vmware.com_capabilities.yaml:330]: Correct.

**Review by danniel1205 — APPROVED** [review]
lgtm

**Review by yharish991 — APPROVED** [review]

**Review by yharish991 — APPROVED** [review]
lgtm!

**Pull request merged**